main.go

```diff
@@ -16,102 +16,51 @@ package main
 
 import (
 	"context"
-	"fmt"
-	"net/http"
 	"os"
+	"os/signal"
+	"path/filepath"
 	"strings"
+	"syscall"
 
-	"github.com/gorilla/mux"
-	"github.com/spf13/pflag"
-	cli "github.com/virtual-kubelet/node-cli"
-	"github.com/virtual-kubelet/node-cli/opts"
-	"github.com/virtual-kubelet/node-cli/provider"
-	"github.com/virtual-kubelet/systemk/pkg/system"
-	"github.com/virtual-kubelet/systemk/systemd"
-	"k8s.io/klog/v2"
+	"github.com/pkg/errors"
+	"github.com/virtual-kubelet/systemk/cmd"
+	"github.com/virtual-kubelet/systemk/internal/provider"
+	"github.com/virtual-kubelet/virtual-kubelet/log"
+	"github.com/virtual-kubelet/virtual-kubelet/log/klogv2"
 )
 
 var (
 	buildVersion = "N/A"
 	buildTime    = "N/A"
-	k8sVersion   = "v1.18.4" // Inject this build time by parsing mod.go
+	k8sVersion   = "v1.18.15" // This should follow the version of k8s.io/kubernetes we are importing
 )
 
 func main() {
-	var (
-		certFile string
-		keyFile  string
-		nodeIP   string
-		nodeEIP  string
-		topdirs  []string
-	)
+	ctx, cancel := context.WithCancel(context.Background())
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		<-sig
+		cancel()
+	}()
 
-	flags := pflag.NewFlagSet("client", pflag.ContinueOnError)
-	// these options need to be make inline with k3s or make clear they afffect logs, certfile and keyfile is too short
-	flags.StringVar(&certFile, "certfile", "", "certfile")
-	flags.StringVar(&keyFile, "keyfile", "", "keyfile")
-	flags.StringVarP(&nodeIP, "node-ip", "i", "", "IP address to advertise for node, '0.0.0.0' or not provided to auto-detect")
-	flags.StringVar(&nodeEIP, "node-external-ip", "", "External IP address to advertise for node, '0.0.0.0' or not provided to auto-detect")
-	flags.StringSliceVarP(&topdirs, "dir", "d", []string{"/var"}, "Only allow mounts below these directories")
+	// Setup VK logger.
+	log.L = klogv2.New(map[string]interface{}{"source": "virtual-kubelet"})
 
-	ctx := cli.ContextWithCancelOnSignal(context.Background())
+	// Setup a local logger.
+	logger := klogv2.New(nil)
 
-	o, err := opts.FromEnv()
-	if err != nil {
-		klog.Fatal(err)
+	var opts provider.Opts
+	if err := provider.SetDefaultOpts(&opts); err != nil {
+		logger.Fatalf("failed to process options: %s", err)
```

Comment:
I think logger, can just be `log` or even `klog`

Reply:
I usually don't like having var names that collide with package names but I'm not strong about the name of the logger. Anyways, as mentioned before, I didn't rework logging as part of this PR but #58 instead. I'm fine doing it here.

Reply:
I like short names and `klog`, or `log` is short and to the point

Reply:
Done!